cmd/wharf/root.go

```diff
@@ -129,3 +145,24 @@ func parseLevel(lvl string) (logger.Level, error) {
 		return logger.LevelDebug, fmt.Errorf("invalid logging level: %q", lvl)
 	}
 }
+
+func handleCancelSignals(cancel context.CancelFunc) {
+	<-waitForCancelSignal()
+	log.Info().WithDuration("gracePeriod", cancelGracePeriod).Message("Cancelling build. Press ^C again to force quit.")
+	cancel()
+
+	select {
+	case <-waitForCancelSignal():
```

Comment:
nit: Could we reuse the channel? Also, maybe name it something like `newCancelSignalChan` or something